agent/src/softwarecontaineragent.cpp

```diff
@@ -277,17 +315,54 @@ std::string SoftwareContainerAgent::bindMountFolderInContainer(const uint32_t co
     return "";
 }
 
-void SoftwareContainerAgent::setGatewayConfigs(const uint32_t &containerID, const std::map<std::string, std::string> &configs)
+bool SoftwareContainerAgent::setGatewayConfigs(const uint32_t &containerID,
+                                               const std::map<std::string, std::string> &configs)
+{
+    profilefunction("setGatewayConfigs");
+    GatewayConfiguration parsedConfigs;
+
+    for (auto const &it : configs) {
+        json_error_t error;
+        std::string gwID = it.first;
+        std::string configStr = it.second;
+
+        json_t *jConfigs = json_loads(configStr.c_str(), 0, &error);
+        if (!jConfigs) {
+            log_error() << "Could not parse config while setting gateway config: " << error.text;
+            log_error() << configs;
+            return false;
+        }
+        parsedConfigs[gwID] = jConfigs;
+    }
+    return updateGatewayConfigs(containerID, parsedConfigs);
+}
+
+bool SoftwareContainerAgent::updateGatewayConfigs(const uint32_t &containerID,
+                                                  const GatewayConfiguration &configs)
 {
-    profilefunction("setGatewayConfigsFunction");
+    profilefunction("updateGatewayConfigs");
     SoftwareContainer *container = nullptr;
-    if (checkContainer(containerID, container)) {
-        container->updateGatewayConfiguration(configs);
+    if (!checkContainer(containerID, container)) {
+        log_error() << "Could not update gateway configuration. Container ("
+                    << std::to_string(containerID) <<") does not exist";
+        return false;
     }
+    container->setGatewayConfigs(configs);
+    return true;
 }
 
-bool SoftwareContainerAgent::setCapabilities(const uint32_t &containerID, const std::vector<std::string> &capabilities)
+
+bool SoftwareContainerAgent::setCapabilities(const uint32_t &containerID,
+                                             const std::vector<std::string> &capabilities)
 {
+    for (std::string capId : capabilities) {
+        GatewayConfiguration newConfigs = m_configStore.getGatewayConfigs(capId);
+
+        if (!updateGatewayConfigs(containerID, newConfigs)) {
```

Comment:
Perhaps a "return" here may have been wrong.
Do we want to fail if we can't set the configuration for one capability?